src/seat.c

```diff
@@ -57,6 +57,37 @@ device_type_from_wlr_device(struct wlr_input_device *wlr_input_device)
 	return NON_TOUCH_DEVICE;
 }
 
+/*
+ * Get applicable profile (category) by matching first by name and secondly be
+ * type (e.g. 'touch' and 'non-touch'). If not suitable match is found based on
+ * those two criteria we fallback on 'default'.
+ */
+static struct libinput_category *
+get_category(struct wlr_input_device *device)
+{
+	enum device_type type = device_type_from_wlr_device(device);
```

Comment:
it looks to me that if this could be done in line 79 (split in 2 if c89 required)

Reply:
Good point. Have moved it.